lib/logproto/logproto-text-server.c

```diff
@@ -383,36 +398,12 @@ log_proto_text_server_fetch_from_buffer(LogProtoBufferedServer *s, const guchar
 
   eol = log_proto_text_server_locate_next_eol(self, state, buffer_start, buffer_bytes);
 
-  if (!eol &&
-      ((buffer_bytes == state->buffer_size) ||
-       log_proto_buffered_server_is_input_closed(&self->super)))
+  if (!eol || !log_proto_text_server_extract(self, state, buffer_start, buffer_bytes, eol, msg, msg_len))
     {
-      log_proto_text_server_yield_whole_buffer_as_message(self, state, buffer_start, buffer_bytes, msg, msg_len);
-      goto success;
-    }
-  else if (!eol)
-    {
-      log_proto_text_server_split_buffer(self, state, buffer_start, buffer_bytes);
-      goto exit;
-    }
-  else
-    {
-      if (!log_proto_text_server_extract(self, state, buffer_start, buffer_bytes, eol, msg, msg_len))
-        {
-          if (buffer_bytes == state->buffer_size)
-            {
-              log_proto_text_server_yield_whole_buffer_as_message(self, state, buffer_start, buffer_bytes, msg, msg_len);
-              goto success;
-            }
-          else
-            {
-              log_proto_text_server_split_buffer(self, state, buffer_start, buffer_bytes);
-              goto exit;
-            }
-        }
+      if (!log_proto_text_server_handle_extract_failures(self, state, buffer_start, buffer_bytes, msg, msg_len))
```

Comment:
@presidento noted that this is not a fully equivalent modification. I'll split this commit.